hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -180,13 +180,28 @@ func (scanner *SchemaScanner) GenerateDefinitions(ctx context.Context, schema *g
 	}
 
 	rootStructRef := astmodel.NewStructReference(
-		scanner.idFactory.CreateIdentifier(rootStructName),
+		scanner.idFactory.CreateIdentifier(rootStructName, astmodel.Public),
 		scanner.idFactory.CreateGroupName(rootStructGroup),
 		scanner.idFactory.CreatePackageNameFromVersion(rootStructVersion),
 		false)
 
 	// TODO: make safer:
-	root := astmodel.NewStructDefinition(rootStructRef, nodes.(*astmodel.StructType))
+	var root *astmodel.StructDefinition
+	switch n := nodes.(type) {
+	case *astmodel.DefinitionName:
+		// A root definition name can happen for example in the case of a oneOf of refs which were all deduped.
+		// When this happens we just need to look up the corresponding struct type
+		if def, ok := scanner.FindDefinition(*n); ok {
+			root = def.(*astmodel.StructDefinition) // TODO: is this safe?
```

Comment:
If it's not, this will panic - should you perhaps use the `root, ok := ...` form and return an error if not?